node_modules/oae-discussions/lib/api.discussions.js

```diff
@@ -508,62 +516,70 @@ var shareDiscussion = module.exports.shareDiscussion = function(ctx, discussionI
+                        roleChanges[principalId] = DiscussionsConstants.roles.MEMBER;
+                    });
+
+                    AuthzAPI.updateRoles(discussionId, roleChanges, function(err) {
                         if (err) {
-                            log().warn({
-                                'err': err,
-                                'principalIds': principalIds,
-                                'discussionId': discussion.id
-                            }, 'Error touching the discussion while sharing. Ignoring.');
+                            return callback(err);
                         }
 
-                        // Use the most recent discussion object
-                        discussion = touchedDiscussion || discussion;
-
-                        _insertLibrary(principalIds, discussion, function(err) {
+                        // Update the discussion `lastModified` date if it has not been updated within a designated threshold. This constitutes an
+                        // update that is by-passing discussion permissions, but that is ok since it is only the `lastModified` date as a result
+                        // of this share
+                        OaeUtil.invokeIfNecessary(_testDiscussionUpdateThreshold(discussion), _touch, discussion, function(err, touchedDiscussion) {
                             if (err) {
                                 log().warn({
                                     'err': err,
                                     'principalIds': principalIds,
                                     'discussionId': discussion.id
-                                }, 'Error updating the library index for these users. Ignoring the error, but some repair may be necessary for these users.');
+                                }, 'Error touching the discussion while sharing. Ignoring.');
                             }
 
-                            DiscussionsAPI.emit(DiscussionsConstants.events.UPDATED_DISCUSSION_MEMBERS, ctx, discussion, roleChanges, principalIds);
-                            return callback();
+                            // Use the most recent discussion object
+                            discussion = touchedDiscussion || discussion;
+
+                            _insertLibrary(principalIds, discussion, function(err) {
+                                if (err) {
+                                    log().warn({
+                                        'err': err,
+                                        'principalIds': principalIds,
+                                        'discussionId': discussion.id
+                                    }, 'Error updating the library index for these users. Ignoring the error, but some repair may be necessary for these users.');
+                                }
+
+                                DiscussionsAPI.emit(DiscussionsConstants.events.UPDATED_DISCUSSION_MEMBERS, ctx, discussion, roleChanges, principalIds);
```

Comment:
Yes, we typically don't discern a difference between sharing and updating members